### PR TITLE
#Sql-tasks. Ivan Menshov

### PR DESCRIPTION
**Pull request by IvanMenshov**
https://travis-ci.com/github/IvanMenshov/db-assignments

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -154,7 +241,17 @@ async function task_1_10(db) {
  *
  */
 async function task_1_11(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+	        ProductName,
+            UnitPrice
+        FROM Products
+        WHERE UnitPrice >= 5 and UnitPrice <16
```
Comment (AlexMokin) [inline task/sql-tasks.js]: that's alright. There is also **between** operator

File: task/sql-tasks.js [hunk 2]
```diff
@@ -178,7 +287,16 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT
```
Comment (AlexMokin) [inline task/sql-tasks.js:291]: it's better to use 2 select statements and one select without from clause. From - clause is optional
eg
```
select
 (select count(*) from ....) as 'Total'
 (select count(*) from ....) as 'TotalDisc'
```

File: task/sql-tasks.js [hunk 3]
```diff
@@ -281,7 +492,26 @@ async function task_1_21(db) {
...
+            d.UnitPrice as 'PricePerItem'
+        FROM Customers
+        INNER JOIN Orders o ON Customers.CustomerID = o.CustomerID
+        INNER JOIN OrderDetails d ON o.OrderID = d.OrderID
+        INNER JOIN Products p ON p.ProductID = d.ProductID
+        WHERE d.UnitPrice = (SELECT MAX(d.UnitPrice)
+            FROM Customers c
+            INNER JOIN Orders o ON Customers.CustomerID = o.CustomerID
+	        INNER JOIN OrderDetails d ON o.OrderID = d.OrderID
+            WHERE Customers.CompanyName = c.CompanyName)
```
Comment (AlexMokin) [inline task/sql-tasks.js]: company name doesn't uniquely identify customer according to the schema. Also company name doens't have an index, so this search is slow

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments. 
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IvanMenshov** [comment]
https://travis-ci.com/github/IvanMenshov/db-assignments

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**